app/src/main/java/com/postpc/tenq/ui/activities/ExistingRoomsActivity.java

```diff
@@ -154,31 +163,125 @@ private void initRecyclerViewWithExistingRooms(List<Room> rooms) {
         IRoomActionListener actionListener = new IRoomActionListener() { //TODO implement
             @Override
             public void onRoomExport(Room room) {
-                Toast.makeText(ExistingRoomsActivity.this, "export clicked!", Toast.LENGTH_SHORT).show();
+                HashMap<String, Object> bodyDetails = new HashMap<>(1);
+                bodyDetails.put("public", false);
+                SpotifyClient
+                        .getClient()
+                        .followPlaylist(room.getPlaylist().getId(), bodyDetails)
+                        .enqueue(new Callback<Void>() {
+                            @Override
+                            public void onResponse(@NotNull Call<Void> call, @NotNull Response<Void> response) {
+                                Toast.makeText(ExistingRoomsActivity.this, "You can now find the playlist on spotify library", Toast.LENGTH_SHORT).show();
+                            }
+
+                            @Override
+                            public void onFailure(@NotNull Call<Void> call, @NotNull Throwable t) {
+                                Log.e("ExistingRoomsActivity", t.getMessage(), t);
+                            }
+                        });
             }
 
+
             @Override
             public void onRoomDelete(Room room) {
                 Toast.makeText(ExistingRoomsActivity.this, "delete clicked!", Toast.LENGTH_SHORT).show();
+                User curUser = getAuthService().getCurrentUser();
+                deleteRoomForUser(room.getId(), curUser);
             }
 
             @Override
             public void onRoomClose(Room room) {
-                Toast.makeText(ExistingRoomsActivity.this, "close clicked!", Toast.LENGTH_SHORT).show();
+                if (!room.getHost().getId().equals(getAuthService().getCurrentUser().getId())){
```

Comment:
Let's hide the "close" button for guests, so it's not even an option? I think it's a better experience :)